### PR TITLE
Make framework::run a method of Configuration

### PR DESCRIPTION
**Pull request by Saethox**
Just a small technicality, but `Configuration` always struck me as somewhat useless after the rework. Making `run` a method of `Configuration` gives it a real purpose, and just flows better, in my opinion.

Maybe we could also rename `run` to `solve`, `optimize`, or `search`, as a `Configuration` doesn't really “run” a problem/ solution space.

### CODE REVIEW

**Comment by luleyleo** [comment]
I like the idea of `Configuration::optimize(problem: P)` and maybe a separate `Configuration::optimize_with(problem: P, init_state: impl FnOnce(state: &mut State))` where the latter would allow setting a custom random generator and other potential state modifications.

@HeleNoir do you have any name preference?

**Comment by HeleNoir** [comment]
I think `optimize` and `optimize_with` are perfect!

**Review by HeleNoir — APPROVED** [review]

File: src/framework/configuration.rs [hunk 1]
```diff
@@ -36,6 +37,47 @@ impl<P: Problem> Configuration<P> {
...
+        state.insert(Random::default());
+        state.insert(state::common::Population::<P>::new());
+
+        heuristic.initialize(problem, &mut state);
+        heuristic.execute(problem, &mut state);
+
+        state
+    }
+
+    pub fn optimize_with(
```
Comment (luleyleo) [inline src/framework/configuration.rs:63]: Documentation is missing. Also `optimize` should mention `optimize_with`.

Comment (Saethox) [reply src/framework/configuration.rs:70]: Better?

Comment (luleyleo) [reply src/framework/configuration.rs:70]: Looks great, thanks!

**Review by luleyleo — APPROVED** [review]

**Pull request merged**